src/RequestParser.php

```diff
@@ -118,4 +85,38 @@ protected function inlineFiles(Request $request): array
 
         return $operations;
     }
+
+    /**
+     * Extracts the body parameters from the request.
+     *
+     * @return array<mixed>
+     */
+    protected function bodyParams(Request $request): array
+    {
+        /**
+         * Never null, since Symfony defaults to application/x-www-form-urlencoded.
+         *
+         * @var string $contentType
+         */
+        $contentType = $request->header('Content-Type');
+
+        if (Str::startsWith($contentType, 'multipart/form-data')) {
+            return $this->inlineFiles($request);
+        }
+
+        $bodyParams = $request->input();
+
+        if (is_array($bodyParams) && Arr::isAssoc($bodyParams)) {
```

Comment:
Should we throw if this is a list?

Reply:
As shown by test `testNonArrayJson()`

Calling 
```php
$request = $this->makeRequest(
    'POST', [], [],
    ['Content-Type' => 'application/json'],
    '"this should be a map with query, variables, etc."'
);
```

Will result in a $bodyParams containing :
```php
array:1 [
  0 => "this should be a map with query, variables, etc."
]
```

Reply:
I think in that case we should definitely throw! Previously, this would have resulted in `'GraphQL Server expects JSON object or array, but got ...`, much better than the generic error message that is now thrown. We should probably extend the tests to ensure that sensical error messages are produced in those edge cases.

Reply:
I added two explicit exception to cover previous scope. Honestly not a big fan at all to provide back to client the given request payload, somehow this could lead to security issue, xss injection etc..